crypto.go

```diff
@@ -26,12 +25,16 @@ func getKey(key string) []byte {
 	}
 	return decodedkey
 }
-func encryptDocs(hits gjson.Result, stream cipher.Stream, fr io.Writer) {
-	var res strings.Builder
-	fmt.Fprintf(&res, "%s\n", hits.Raw)
-	plainText := []byte(res.String())
-	cipherText := make([]byte, len(plainText))
-	stream.XORKeyStream(cipherText, plainText)
+
+func encryptDocs(buf *bytes.Buffer, stream cipher.Stream, fr io.Writer) {
+	plainText := base64.StdEncoding.EncodeToString(buf.Bytes())
+
+	resn := new(strings.Builder)
+	fmt.Fprintf(resn, "%s%%", plainText)
+	plainTextDel := resn.String()
+
+	cipherText := make([]byte, len(plainTextDel))
+	stream.XORKeyStream(cipherText, []byte(plainTextDel))
```

Comment:
I really don't understand what's going on here.

Reply:
Nothing too fancy. I append a delimiter to each of the compressed data blocks and then the result of that is encrypted.

Reply:
Why do we want a delimeter?

Also, base64-encoding feels counterproductive to compression.

Reply:
It is used to process each batch of documents at a time. The reson for b64 is that it has a limited character set. This allows us to choose a delimiter without the risk of bad partitioning of the data blocks.

Reply:
I think it's a bit expensive for that benefit. Especially if we do non-streaming I'd much prefer having a length field before each data chunk.

But since we already rely on the data being json, I don't see a reason we can't do a stream and pick each object off without delimeters, just reading each json object. It's at least possible (although not entirely straightforward) with the built in json library.

Reply:
Agreed! We can look into this for the next iteration.

Reply:
> But since we already rely on the data being json, I don't see a reason we can't do a stream and pick each object off without delimeters, just reading each json object. It's at least possible (although not entirely straightforward) with the built in json library.

Worst case scenario, you have to unmarshal the json blob into an struct interface and iterate through that.